nanoFramework.Json/TimeExtensions.cs

```diff
@@ -109,15 +137,47 @@ public static DateTime FromIso8601(string date)
 			string second = (parts.Length > 5) ? parts[5] : "0";
 			string ms = (parts.Length > 6) ? parts[6] : "0";
 
+			// If the input string is non-numeric, Convert.ToInt32() below will throw an exception from the Native code
+			// These exceptions cause annoying delays & output when running in the nanoFramework debugger
+			// Bail out if the input string is non-numeric
+			if (!IsNumeric(year))
+			{
+				return DateTime.MaxValue;
+			} 
+			else if (!IsNumeric(month))
+			{
+				return DateTime.MaxValue;
+			} 
+			else if (!IsNumeric(day))
+			{
+				return DateTime.MaxValue;
+			} 
+			else if (!IsNumeric(hour))
+			{
+				return DateTime.MaxValue;
+			} 
+			else if (!IsNumeric(minute))
+			{
+				return DateTime.MaxValue;
+			} 
+			else if (!IsNumeric(second))
+			{
+				return DateTime.MaxValue;
+			} 
+			else if (!IsNumeric(ms))
+			{
+				return DateTime.MaxValue;
+			}
+
 			// sanity check for bad milliseconds format
 			int milliseconds = Convert.ToInt32(ms);
 
-			if(milliseconds > 999)
+            if (milliseconds > 999)
             {
 				milliseconds = 999;
 			}
 
-			DateTime dt = new(
+            DateTime dt = new(
```

Comment:
same here for the alignment

Reply:
Done.  There's no visible difference between the old and new.  Is it correct now?